src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -6,29 +6,14 @@
 
 using System.Diagnostics;
 using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax
 {
     internal partial class LanguageParser
     {
-        /// <summary>
-        /// "Safe" substring using start and end positions rather than start and length.
```

Comment:
First, 'safe' is crazy here.  We should never be in a situation where we need to munge positions.  Everything shoudl be known with exact locations and we should always be able to safely use the data collected.  This approach to specifying inclusive end positions, but then rectifying them is just bad.